kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtNetworkServiceImpl.java

```diff
@@ -1104,6 +1108,41 @@ public void updateNetInterfaceConfigurations(GwtXSRFToken xsrfToken, GwtNetInter
         return new ArrayList<>(gwtWifiHotspotsEntries);
     }
 
+    @Override
+    public List<GwtModemPdpEntry> findPdpContextInfo(GwtXSRFToken xsrfToken, String interfaceName)
+            throws GwtKuraException {
+        checkXSRFToken(xsrfToken);
+
+        List<GwtModemPdpEntry> gwtModemPdpEntries = new ArrayList<>();
+        ModemManagerService mms = ServiceLocator.getInstance().getService(ModemManagerService.class);
+        CellularModem modem = getModemService(interfaceName, mms);
+        if (modem == null) {
+            return gwtModemPdpEntries;
+        }
+        try {
+            List<ModemPdpContext> pdpContextInfo = modem.getPdpContextInfo();
+            GwtModemPdpEntry gwtModemPdpEntry;
+            int contextNo = 1;
+            for (ModemPdpContext pdpContextEntry : pdpContextInfo) {
+                gwtModemPdpEntry = new GwtModemPdpEntry();
+                gwtModemPdpEntry.setContextNumber(pdpContextEntry.getNumber());
+                gwtModemPdpEntry.setPdpType(pdpContextEntry.getType().getValue());
+                gwtModemPdpEntry.setApn(pdpContextEntry.getApn());
+                gwtModemPdpEntries.add(gwtModemPdpEntry);
+                contextNo++;
+            }
+            gwtModemPdpEntry = new GwtModemPdpEntry();
```

Comment:
why this last entry?

Reply:
This is needed to create new PDP context.